drafts/accept_confirm.md

```diff
@@ -0,0 +1,67 @@
+# You're not stubbing, stupid!
+
+Recently, in a [Ruby on Rails](https://rubyonrails.org/) project, I was writing a [Cucumber](https://cucumber.netlify.app/docs/installation/ruby/) scenario that was deleting a resource by having the user clicking a 'Delete' button. Before the action was executed, the user had to confirm a message shown in a confirmation dialog. You may have seen this dozens of times when scaffolding a Rails application.  
+Oh and upon deleting, I also had to do a request to an external API (to be more precise: the use case was that a user unsubscribing, so I had to send a DELETE request to a [Mollie API](https://docs.mollie.com/reference/v2/subscriptions-api/cancel-subscription)).
+
+So, "nothing new here", I thought. I knew about [WebMock](https://github.com/bblimke/webmock), since I wanted to stub the external API request and my test suite was set up to test [JavaScript](https://github.com/teamcapybara/capybara#setup), so I knew I could use the [`accept_confirm`](https://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FSession:accept_confirm) method here that Capybara offers. 
+
+Stubbing the request was defined in a support file:
+
+```ruby
+# features/support/webmock.rb
+
+require 'webmock/cucumber'
+
+WebMock.disable_net_connect!(allow_localhost: true)
+
+Before do |_scenario|
+  stub_request(
+    :delete, 
+    %r{https://api.mollie.com/v2/customers/\w+/subscriptions/\w+}
+  )
+    .to_return(body: {}.to_json)
+end
+
+```
+
+My step implementation looked like this:
+
+```ruby
+# features/step_definitions/general_steps.rb
+
+When('I delete the resource') do
+  accept_confirm { click_on 'Destroy' }
```

Comment:
For reference:
https://www.rubydoc.info/github/jnicklas/capybara/Capybara%2FSession:accept_confirm

1. Execute the block, accepting a confirm.
2. Expects a block whose actions will trigger the display modal to appear.

Which Javascript Driver are you using? 

Maybe the accept_modal is not implemented in that driver.

I also remember that (most of the) drivers ignore any native dialog boxes (confirm, alert) by default.

```
# https://www.rubydoc.info/github/jnicklas/capybara/Capybara/Selenium/Driver#accept_modal-instance_method
def accept_modal(_type, **options)
  yield if block_given?
  modal = find_modal(options)

  modal.send_keys options[:with] if options[:with]

  message = modal.text
  modal.accept
  message
end
```

When using`options[:with]` and then use `:enter` for the `modal.send_keys`, that might also work.

Reply:
The problem was not that the confirmation dialog wasn't accepted (so the `accept_modal` worked), but the `stub_request` was not recognized. But maybe that wasn't clear from the article?